src/tools/pathTools.cpp

```diff
@@ -304,18 +304,14 @@ std::string kiwix::getFileContent(const std::string& path)
 
 bool kiwix::fileExists(const std::string& path)
 {
-#ifdef _WIN32
-  return PathFileExistsW(Utf8ToWide(path).c_str());
-#else
-  bool flag = false;
-  std::fstream fin;
-  fin.open(path.c_str(), std::ios::in);
-  if (fin.is_open()) {
-    flag = true;
-  }
-  fin.close();
-  return flag;
-#endif
+  std::fstream fin(path.c_str());
+  return !fin.fail();
```

Comment:
Is this working if the file exists but don't have read right ?

Reply:
Yes
![image](https://user-images.githubusercontent.com/26814569/148068183-3ee8fab9-6ebf-4393-b26b-31e02c43e603.png)

Reply:
I don't know the code of `app` but it seems it return different result depending if the file is readable or not.
But as we use this check in `fileExists` it should return the same result.

Reply:
Oops, that was a bad screenshot, my bad.
![image](https://user-images.githubusercontent.com/26814569/148194458-0302eec6-ee77-4767-9c20-ce5e9cad6d2d.png)

app's code:

```
#include <fstream>
#include <iostream>
int main()
{
  std::string path = "lol.txt";
  std::fstream fin(path.c_str());
  std::cout << !fin.fail() << std::endl;
  return 0;
}
```

Reply:
Oh, correct. I got confused.

Reply:
> About Windows code, what is the way to check if a file is readable? 

It is not so easy. I've found :
 - [GetNamedSecurityInfoW](https://docs.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-getnamedsecurityinfowhttps://docs.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-getnamedsecurityinfow) but it seems pretty complex to use and I'm not even sure it provides what we need
 - [CreateFile](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew) Use with the right option, it doesn't create a file but create a handle to a existing file if it exist and has the right needed. It is somehow the same that creating a c++ stream, but as c++ stream creators have no version for widestring, we need to use specific windows functions (https://stackoverflow.com/a/2112336/292425)

To check that a file exists (and just exists, whatever the rights we have). The best is probably to use [`stat`](https://www.man7.org/linux/man-pages/man2/stat.2.html) ([`_wstat`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions?view=msvc-170) on windows)

Reply:
I'll check those, https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/access-waccess?view=msvc-170 this looks interesting also